.github/actions/triggerWorkflowAndWait/triggerWorkflowAndWait.js

```diff
@@ -0,0 +1,138 @@
+        })
+
+        .catch((err) => {
+            console.error(`Failed to dispatch workflow ${workflow}`, err);
+            core.setFailed(err);
+            process.exit(1);
+        })
+
+        // Wait for the new workflow to start
+        .then(() => {
+            let waitTimer = -POLL_RATE;
+            return promiseWhile(
+                () => !hasNewWorkflowStarted && waitTimer < NEW_WORKFLOW_TIMEOUT,
+                _.throttle(
+                    () => {
+                        console.log(`\n🤚 Waiting for a new ${workflow} workflow run to begin...`);
+                        return githubUtils.getLatestWorkflowRunID(workflow)
+                            .then((lastWorkflowRunID) => {
+                                newWorkflowRunID = lastWorkflowRunID;
+                                hasNewWorkflowStarted = newWorkflowRunID !== previousWorkflowRunID;
+
+                                if (!hasNewWorkflowStarted) {
+                                    waitTimer += POLL_RATE;
+                                    if (waitTimer < NEW_WORKFLOW_TIMEOUT) {
+                                        // eslint-disable-next-line max-len
+                                        console.log(`After ${waitTimer / 1000} seconds, there's still no new ${workflow} workflow run 🙁`);
+                                    } else {
+                                        // eslint-disable-next-line max-len
+                                        const err = new Error(`After ${NEW_WORKFLOW_TIMEOUT} seconds, the ${workflow} workflow did not start.`);
+                                        console.error(err);
+                                        core.setFailed(err);
+                                        process.exit(1);
+                                    }
+                                } else {
+                                    console.log(`\n🚀 New ${workflow} run with ID ${newWorkflowRunID} has started`);
+                                }
+                            })
+                            .catch((err) => {
+                                console.warn('Failed to fetch latest workflow run.', err);
+                            });
+                    },
+                    POLL_RATE,
+                ),
+            );
+        })
+
+        // Wait for the new workflow run to finish
+        .then(() => promiseWhile(
+            () => !workflowCompleted,
+            _.throttle(
+                () => {
+                    console.log(`\n⏳ Waiting for workflow run ${newWorkflowRunID} to finish...`);
+                    return octokit.actions.getWorkflowRun({
+                        owner: GithubUtils.GITHUB_OWNER,
+                        repo: GithubUtils.EXPENSIFY_CASH_REPO,
+                        run_id: newWorkflowRunID,
+                    })
+                        .then(({data}) => {
+                            workflowCompleted = data.status === 'completed' && data.conclusion !== null;
```

Comment:
Can the workflow get stuck and never moved to a "completed" status? If so, what would it happen? Would we keep polling forever for the status?

Reply:
> Can the workflow get stuck and never moved to a "completed" status?

Not that I know of, but it might be possible.

> Would we keep polling forever for the status?

Theoretically if we started waiting for a GitHub workflow that got stuck, we would continue polling for the status until the maximum job runtime of 6 hours is reached, at which point GitHub would kill the job.

Reply:
Do we want to kill it sooner? Or it's ok if we leave polling until we reach the max job runtime?

Reply:
6 hours seems like a long time to wait in this case, have we had any workflows that have gotten stuck in the `queued` state? Also, what is the longest amount of time we'd expect a workflow to take? If something is hanging longer than we expect it to, I think we'd want to get it resolved sooner than later, so I think something like 2 hours is more appropriate here (assuming there aren't any workflows that take that long atm).

Reply:
> have we had any workflows that have gotten stuck in the queued state

I have never seen this happen, no.

 > something like 2 hours is more appropriate here 
 
 Okay, yeah the longest-running workflow we have is the iOS deploy, which usually takes about 30 minutes. So I can add a manual timeout and alert if we reach that timeout.

Reply:
Eerie coincidence ... for the first time I've ever seen it, it appears the iOS E2E test hung on this PR and gave up after 6 hours. I guess I should take that as a sign 😄

Reply:
Okay, I added a workflow execution timeout of 2 hours